crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -168,13 +163,22 @@ impl AuthorityPerpetualTables {
 
     pub fn get_recovery_epoch_at_restart(&self) -> SuiResult<EpochId> {
         Ok(self
-            .current_epoch
-            .get(&CURRENT_EPOCH_KEY)?
-            .expect("Must have current epoch."))
+            .epoch_start_configuration
+            .get(&())?
+            .expect("Must have current epoch.")
+            .epoch_id())
     }
 
-    pub fn set_recovery_epoch(&self, epoch: EpochId) -> SuiResult {
-        self.current_epoch.insert(&CURRENT_EPOCH_KEY, &epoch)?;
+    pub async fn set_epoch_start_configuration(
+        &self,
+        epoch_start_configuration: &EpochStartConfiguration,
+    ) -> SuiResult {
+        let mut wb = self.epoch_start_configuration.batch();
```

Comment:
qq - why do we use batch here? (since it's a single write anyway and we commit batch in next line)

Reply:
This is mostly because we are soon getting rid of single put() api in typed-store and only going to support write batch as a single way to update db. That's why I moved it to write batch to incur less work in future.